opensoundscape/torch/models/utils.py

```diff
@@ -240,13 +241,22 @@ def tensor_binary_predictions(scores, mode, threshold=None):
         if threshold is None:
             raise ValueError(f"threshold must be specified for multi_target prediction")
         # predict 0 or 1 based on a fixed threshold
-        preds = torch.sigmoid(scores) >= threshold
+        try:
+            if len(threshold) == 1 or len(threshold) == len(scores[0]):
+                preds = torch.sigmoid(scores) >= torch.tensor(threshold)
```

Comment:
it's a bit surprising to me that this line actually handles the multiple threshold case correctly, so I'd add a one-line comment just stating that it will do per-class prediction if there's a list of thresholds.